components/layout/query.rs

```diff
@@ -797,18 +847,20 @@ pub fn process_offset_parent_query<N: LayoutNode>(requested_node: N, layout_root
         -> OffsetParentResponse {
     let mut iterator = ParentOffsetBorderBoxIterator::new(requested_node.opaque());
     sequential::iterate_through_flow_tree_fragment_border_boxes(layout_root, &mut iterator);
+
+    let node_offset_box = iterator.node_offset_box;
     let parent_info_index = iterator.parent_nodes.iter().rposition(|info| info.is_some());
-    match parent_info_index {
-        Some(parent_info_index) => {
+    match (node_offset_box, parent_info_index) {
+        (Some(node_offset_box), Some(parent_info_index)) => {
             let parent = iterator.parent_nodes[parent_info_index].as_ref().unwrap();
-            let origin = iterator.node_border_box.origin - parent.border_box.origin;
-            let size = iterator.node_border_box.size;
+            let origin = node_offset_box.offset - parent.border_box.origin;
```

Comment:
@stshine Sorry for taking a week to respond, but that's a really great catch. Thank you for looking at this more closely than I did.